website/mails/mails.py

```diff
@@ -117,6 +124,79 @@ def send_mail(to_addr, mail, mimetype='plain', from_addr=None, mailer=None,
 
         return ret
 
+def queue_mail(to_addr, mail, send_at, user, **context):
+    """
+    Queue an email to be sent using send_mail after a specified amount
+    of time and if the callback returns True. The callback is attached to
+    the template under mail.
+
+    :param to_addr: the address email is to be sent to
+    :param mail:  the type of mail. Struct following template:
+                        { 'callback': function(),
+                            'template': mako template name,
+                            'subject': mail subject }
+    :param send_at: datetime object of when to send mail
+    :param user: user object attached to mail
+    :param context: IMPORTANT kwargs to be attached to template.
+                    Sending mail will fail if wrong all kwargs are
+                    not parameters.
+    :return: the QueuedMail object created
+    """
+    new_mail = QueuedMail(
+        user=user,
+        to_addr=to_addr,
+        send_at=send_at,
+        email_type=mail['template'],
+        data=context
+    )
+    new_mail.save()
+    return new_mail
+
+class QueuedMail(StoredObject):
+    _id = fields.StringField(primary=True, default=lambda: str(bson.ObjectId()))
+    user = fields.ForeignField('User')
```

Comment:
Can you add more validation to these fields? For example, if `user` is required, set `required=True`.